File: src/decoders/resource-content.ts
import type { ResourceContent, SessionHandle } from "~/models";
import { decodeAssignmentTheme } from "./assignment-theme";
import { decodeAttachment } from "./attachment";

export const decodeResourceContent = (content: any, session: SessionHandle): ResourceContent => {
  return {
    id: content.N,
    title: content.L,
    description: content.descriptif.V,
    category: content.categorie.V.G,
    files: content.ListePieceJointe.V.map((attachment: any) => decodeAttachment(attachment, session)),
    themes: content.ListeThemes.V.map((theme: any) => decodeAssignmentTheme(theme)),
    // TODO: Investigate to see what is contained here when not `-1`.
    educativeValue: content.parcoursEducatif
  };
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/decoders/resource-content.kt
package ink.literate.pawnote.decoders

import ink.literate.pawnote.models.ResourceContent
import ink.literate.pawnote.models.ResourceContentCategory
import ink.literate.pawnote.models.SessionHandle
import kotlinx.serialization.json.*

fun decodeResourceContent (content: JsonObject, session: SessionHandle) = ResourceContent(
    id = content["N"]!!.jsonPrimitive.content,
    title = content["L"]!!.jsonPrimitive.content,
    description = content["descriptif"]!!.jsonObject["V"]!!.jsonPrimitive.content,
    category = ResourceContentCategory.fromInt(content["categorie"]!!.jsonObject["V"]!!.jsonObject["G"]!!.jsonPrimitive.int),
    files = content["ListePieceJointe"]!!.jsonObject["V"]!!.jsonArray.map { decodeAttachment(it.jsonObject, session.information) },
    themes = content["ListeThemes"]!!.jsonObject["V"]!!.jsonArray.map { decodeAssignmentTheme(it.jsonObject) },
    // TODO: Investigate to see what is contained here when not `-1`.
    educativeValue = content["parcoursEducatif"]!!.jsonPrimitive.int
)